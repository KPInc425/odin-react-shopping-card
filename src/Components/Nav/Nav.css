nav {
    background: linear-gradient(280deg, rgb(135, 173, 139) 20%, rgb(164, 214, 169) 40%, rgb(197, 238, 197) 100%);

    /* margin-bottom: 20px; */
    padding: 10px;
    display: flex;
    justify-content: space-between;
    border-bottom: 10px double rgb(0, 0, 0);
    position: sticky;
    top: 0;
    z-index: 10;
}

nav ul {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    margin-right: 10px;
}

li {
    list-style: none;
    margin: 10px;
    font-size: large;
}

.link {
    text-decoration: none;
    color: white;
    padding: 5px;
}



.link:active {
    background-color: rgba(0, 0, 0, 0.425);
    /* padding: 5px; */
    border-radius: 12px;
}

.link:hover li {
    font-size: 150%;
}

nav img {
    width: 64px;
    height: 64px;
}

@media (max-height: 650px) {
    nav {
        position: relative;
    }
}