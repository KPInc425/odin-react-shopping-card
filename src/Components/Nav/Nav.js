import { Link } from 'react-router-dom';
import './Nav.css';

const Nav = () => {
    return (
        <nav>
            <img src={ `${process.env.PUBLIC_URL}/ilg128.png` } alt='Logo'></img> 
            <ul>
                <Link to='/' className='link'>
                    <li>Home</li>
                </Link>
                <Link to='/shop' className='link'>
                    <li>Shop</li>
                </Link>
            </ul>
        </nav>
    )
}

export default Nav;