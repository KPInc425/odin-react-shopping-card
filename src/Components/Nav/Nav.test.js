import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Nav from './Nav';

describe("Nav Component", () => {
    it("render nav component", () => {
        const { container } = render(<BrowserRouter>
                                        <Nav />
                                    </BrowserRouter>);
        expect(container).toMatchSnapshot();
    });

    it("has links to homepage and shoppage", () => {
        render(<BrowserRouter>
            <Nav />
        </BrowserRouter>);
        const homeLink = screen.getByRole("link", { name: "Home" });
        const shopLink = screen.getByRole("link", { name: "Shop" });

        const homeLinkRoute = homeLink.href.split("localhost");
        const shopLinkRoute = shopLink.href.split("localhost");

        expect(homeLinkRoute[1]).toMatch('/');
        expect(shopLinkRoute[1]).toMatch('/shop');
    })

})
