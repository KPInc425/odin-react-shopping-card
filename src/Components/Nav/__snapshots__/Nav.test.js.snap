// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component render nav component 1`] = `
<div>
  <nav>
    <img
      alt="Logo"
      src="/favicon.ico"
    />
    <ul>
      <a
        class="link"
        href="/"
      >
        <li>
          Home
        </li>
      </a>
      <a
        class="link"
        href="/shop"
      >
        <li>
          Shop
        </li>
      </a>
    </ul>
  </nav>
</div>
`;
